### PR TITLE
Article refactoring

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

I originally came to this model to move some things into the decorator, but that turned into a dead end since several methods were calling into each other and with some of them also being called in the search serializer it was too hard to untangle them. So instead I decided to just clean up some code.

## Related Tickets & Documents

Cooldown fun.

## QA Instructions, Screenshots, Recordings

Nothing specific. This is a refactoring, so I just make sure that we have specs for code I'm changing and that they stay green.

## Added tests?

- [ ] yes
- [X] no, because they aren't needed (see above)
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [X] no documentation needed

### CODE REVIEW

File: app/models/article.rb [hunk 1]
```diff
@@ -355,13 +353,11 @@ def cloudinary_video_url
   end
 
   def video_duration_in_minutes
-    minutes = video_duration_in_minutes_integer
-    seconds = video_duration_in_seconds.to_i % 60
-    seconds = "0#{seconds}" if seconds.to_s.size == 1
+    duration = ActiveSupport::Duration.build(video_duration_in_seconds.to_i).parts
```
Comment (citizen428) [inline app/models/article.rb:356]: This takes a value in seconds and turns it into a useful hash:

```ruby
ActiveSupport::Duration.build(4567).parts
#=> { hours: 1, minutes: 16, seconds: 7 }
```

File: app/models/article.rb [hunk 2]
```diff
@@ -355,13 +353,11 @@ def cloudinary_video_url
...
+    minutes_and_seconds = format("%<minutes>02d:%<seconds>02d", duration)
```
Comment (citizen428) [inline app/models/article.rb:357]: This is standard Ruby string formatting, but using some of the more advanced options:

1. `"%02d"` means "take a number and format it as a 2 digit string, with leading zero added if necessary".
    ```ruby
    format("%02d", 1)
    #=> "01"
    format("%02d", 12)
    #=> "12"
    ```
2. Apart from individual arguments, format also allows a hash. Names in angle brackets refer to hash keys, so  `"%<minutes>02d` formats the value of `duration[:minutes]` as described above.

**Comment by citizen428** [comment]
I guess we can ignore CodeClimate, as the 2 errors have nothing to do with the contents of this PR.

**Comment by rhymes** [comment]
@citizen428 I guess brakeman got updated on their end, I wonder if we could fix them nonetheless because I reckon they are going to appear on contributor's PRs at some point as well.

I believe we can "fast track" this PR

File: app/models/article.rb [hunk 3]
```diff
@@ -355,13 +353,11 @@ def cloudinary_video_url
...
+    return minutes_and_seconds if duration[:hours] < 1
 
-    hours = (video_duration_in_seconds.to_i / 3600)
-    minutes = "0#{minutes}" if hours.positive? && minutes < 10
-    hours < 1 ? "#{minutes}:#{seconds}" : "#{hours}:#{minutes}:#{seconds}"
+    "#{duration[:hours]}:#{minutes_and_seconds}"
   end
 
   def video_duration_in_minutes_integer
```
Comment (rhymes) [inline app/models/article.rb:363]: can we move this in the serializer? It's only called there... The serializer already contains some serialization logic.

**Review by rhymes — APPROVED** [review]
Left a note about moving `video_duration_in_minutes_integer` but other than that 🆗

**Review by benhalpern — APPROVED** [review]

File: app/models/article.rb (hunk 3 above)
Comment (citizen428) [reply app/models/article.rb:363]: Since this got merged already, I'll check that tomorrow.

**Pull request merged**